src/routes/+page.svelte

```diff
@@ -1,2 +1,93 @@
-<h1>Welcome to SvelteKit</h1>
-<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>
+<script>
+	import { createEventDispatcher } from 'svelte';
+
+	let url = '';
+	let shortUrl = '';
+	let error = '';
+	let loading = false;
+	let success = false;
+
+	const dispatch = createEventDispatcher();
+
+	function validateUrl(value) {
+		try {
+			new URL(value);
+			return true;
+		} catch (_) {
+			return false;
+		}
+	}
+
+	async function handleSubmit(event) {
+		event.preventDefault();
+		error = '';
+		success = false;
+
+		if (!validateUrl(url)) {
+			error = 'Invalid URL';
+			return;
+		}
+
+		loading = true;
```

Comment:
Better use int++ \ --
it will protect us from a race if there is a list of requests in one moment